experiments/AMIP/user_io/leaderboard/utils.jl

```diff
@@ -0,0 +1,205 @@
+"""
+    isequispaced(arr::Vector)
+
+Return whether the array is equispaced or not.
+"""
+function isequispaced(arr::Vector)
+    return all(diff(arr) .≈ arr[begin + 1] - arr[begin])
+end
+
+"""
+    resample(
+        data::AbstractArray,
+        src_lonlat::Tuple{<:AbstractArray, <:AbstractArray},
+        dest_lonlat::Tuple{<:AbstractArray, <:AbstractArray},
+        )
+
+Resample 2D `data` from `src_lonlat` to `dest_lonlat`.
+
+Note, "flat" boundary conditions are imposed for the outer edges. This should make sense for
+most data sources (that are defined all over the globe but do not necessarily have points at
+the poles). Be sure to check that it makes sense for your data.
+
+The resampling performed here is a 0th-order constant resampling.
+"""
+
+function resample(
+    data::AbstractArray,
+    src_lonlat::Tuple{<:AbstractArray, <:AbstractArray},
+    dest_lonlat::Tuple{<:AbstractArray, <:AbstractArray},
+)
+
+    # Interpolations.jl wants ranges, so we have to make ranges out of the given src_lonlat
+    #
+    # NOTE: We are assuming that lonlat are equispaced. Check this!
+    vec_to_range(v) = range(v[begin], v[end], length = length(v))
+    src_lonlat_ranges = vec_to_range.(src_lonlat)
+
+    itp = Interpolations.constant_interpolation(src_lonlat_ranges, data, extrapolation_bc = Interpolations.Flat())
+
+    dest_lon, dest_lat = dest_lonlat
+
+    return [itp(lon, lat) for lon in dest_lon, lat in dest_lat]
+end
+
+"""
+    integration_weights(lonlat::Tuple{<:AbstractArray, <:AbstractArray})
+
+Compute the integration weights for a first-order spherical integration.
+"""
+function integration_weights(lonlat::Tuple{<:AbstractArray, <:AbstractArray})
+    lon, lat = lonlat
+
+    # We are also assuming that they are in degrees
+    abs(maximum(lon)) <= π && error("longitude should be in degrees")
+    abs(maximum(lat)) <= π / 2 && error("latitude should be in degrees")
+
+    isequispaced(lon) || error("Longitude is not equispaced")
+    isequispaced(lat) || error("Latitude is not equispaced")
+
+    dlon_rad = deg2rad(abs(lon[begin + 1] - lon[begin]))
+    dlat_rad = deg2rad(abs(lat[begin + 1] - lat[begin]))
+
+    return [cosd(lat1) * dlon_rad * dlat_rad for _ in lon, lat1 in lat]
+end
+
+"""
+    integrate_on_sphere(
+        data::AbstractArray,
+        lonlat::Tuple{<:AbstractArray, <:AbstractArray},
+        )
+
+Integrate `data` on a sphere with a first-order scheme. `data` has to be discretized on
+`lonlat`.
+"""
+function integrate_on_sphere(data::AbstractArray, lonlat::Tuple{<:AbstractArray, <:AbstractArray})
+    lon, lat = lonlat
+    size_data = size(data)
+    exp_size = (length(lon), length(lat))
+    size_data == exp_size || error("Inconsistent dimensions $size_data != $exp_size")
+    return sum(data .* integration_weights(lonlat))
```

Comment:
```suggestion
    return sum(data .* integration_weights(lonlat)) ./ sum(interpolation_weights(lonlat))
```
Or is the sum of weights 1 here?

Reply:
They are 1 in the limit of infinite resolution. It's a good idea to normalize them